x-pack/plugins/alerts/server/alerts_client.ts

```diff
@@ -796,16 +762,10 @@ export class AlertsClient {
 
     const mutedInstanceIds = attributes.mutedInstanceIds || [];
     if (!attributes.muteAll && mutedInstanceIds.includes(alertInstanceId)) {
-      await this.unsecuredSavedObjectsClient.update(
-        'alert',
-        alertId,
-        {
-          updatedBy: await this.getUserName(),
-
-          mutedInstanceIds: mutedInstanceIds.filter((id: string) => id !== alertInstanceId),
-        },
-        { version }
-      );
+      await partiallyUpdateAlertSavedObject(this.unsecuredSavedObjectsClient, alertId, {
+        updatedBy: await this.getUserName(),
+        mutedInstanceIds: mutedInstanceIds.filter((id: string) => id !== alertInstanceId),
+      });
```

Comment:
Ah, I didn't realize the API really only let's you change a single instance in the `[un]muteInstance()` APIs.  So I think you're right, we may get a race condition with customers clicking on mute/unmute of instances.

In this case, I think we should probably go with a story where we keep the existing OCC calls, but wrapper the relevant alertsClient methods so they do a small, fixed number of retries (eg, 5) with a slight delay (eg, 250ms) in the case that they get 409 conflict errors.  As noted in this comment https://github.com/elastic/kibana/pull/76830#issuecomment-689165423 - I tried out this technique as well and it managed to get the SIEM function tests working, which is where I noticed the problem.

The reason I want to do it that way is that it should avoid having the customers see 409 conflict results because of our updates to the upcoming `execution_status` fields.  Customers would end up having to have their own 409 conflict retry logic around EVERY alert API.

Once we have "version" support for our APIs (or just for `update()`?), we can bypass the retries.

Reply:
++ I like such approach; keeping OCC for now with retry logic. This will be a better experience 👍

The delay could also be a random number between 0 and 250ms in case the API is called at the same time for multiple instances (ex: in a for loop per instance).